### DISCUSSION TITLE
RTSP/VLC/FFPlay silently freezed stream on random frame

### DISCUSSION THREAD

**Discussion by brandousov** [bluenviron/mediamtx#272] [Questions]
## Which version are you using?

v0.13.3

## Question

Hi! We have IP cam's with RTSP streams, proxied by rtsp-simple-server, which freezes on random frame, on random time when I testing it with `VLC` or `ffplay` (no error usually, only few dropped frames sometimes - both on good and frozen cams). Our architecture designed as such a way that we use TCP only protocol.  

That issue happens with 1-2 cameras of 10. Other ("good") cam's freezes too, but for only 1-2 times a day. "Bad" cam's freezes every 1-3-5-10-15-20 minutes randomly, if I restart VLC with same MRL it works fine from start.. How can I detect frozen stream if `ffplay` still shows the same bitrate with same parameters from start without any error?  

Screenshot: https://github.com/brandousov/pix/blob/main/rtsp-freeze.jpg

**Comment by aler9** [comment]
Hello, if cameras are working correctly, freezes are caused by packet loss.
If the protocol is TCP and there aren't reconnections, packet loss is possible only when `readBufferCount` is too low - try increasing it. Otherwise, it's the camera.

[a comment by an automated account is not shown]
